### PR TITLE
fix local volume absolute path issue on Windows

### LINKED ISSUE #62016: local volume does not accept Windows absolute path
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

> Uncomment only one, leave it on its own line: 
>
> /kind bug
> /kind feature

**What happened**:
Windows absolute path `D:` is not allowed in local volume, the [validation](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L1386) happens on both master and agent node, while for windows cluster, the master is Linux and agent is Windows, so `path.IsAbs()` func will not work all in both nodes.
User will get following error when specify `D:` for `spec.local`:
```
The PersistentVolume "pv-local" is invalid: spec.local: Invalid value: "D:": must be an absolute path
```

Complete PV example is here:
```
apiVersion: v1
kind: PersistentVolume
metadata:
  name: pv-local
spec:
  capacity:
    storage: 100Gi
  volumeMode: Filesystem
  accessModes:
  - ReadWriteOnce
  persistentVolumeReclaimPolicy: Delete
  storageClassName: local-storage
  local:
    path: "D:"
  nodeAffinity:
    required:
      nodeSelectorTerms:
      - matchExpressions:
        - key: kubernetes.io/hostname
          operator: In
          values:
          - 20941k8s9010
```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.11
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

/sig windows
/sig storage

### PR DESCRIPTION
**Pull request by andyzhangx**
**What this PR does / why we need it**:
remove IsAbs validation on local volume since it does not work on windows cluster, Windows absolute path `D:` is not allowed in local volume, the [validation](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation.go#L1386) happens on both master and agent node, while for windows cluster, the master is Linux and agent is Windows, so `path.IsAbs()` func will not work all in both nodes. 
**Instead**, this PR use `MakeAbsolutePath` func to convert `local.path` value in kubelet, it supports both linux and windows styple. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #62016

**Special notes for your reviewer**:

**Release note**:

```
fix local volume absolute path issue on Windows
```
/sig storage
/sig windows

### CODE REVIEW

**Comment by msau42** [comment]
I think similarly to #55665, we need to make absolute path on the kubelet (local.go) side, otherwise the user could be allowed to mount a relative path.

/assign

**Comment by andyzhangx** [comment]
@msau42 I have moved `MakeAbsolutePath` func from kubelet_pods.go to volume\utils.go, and use same logic as #55665, PTAL, thanks.

File: pkg/volume/local/local.go [hunk 1]
```diff
@@ -311,7 +311,8 @@ func (m *localVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 	}
 
 	glog.V(4).Infof("attempting to mount %s", dir)
-	err = m.mounter.Mount(m.globalPath, dir, "", options)
+	globalPath := util.MakeAbsolutePath(runtime.GOOS, m.globalPath)
```
Comment (msau42) [inline pkg/volume/local/local.go]: `SetUpDevice()` also needs to be updated

**Comment by msau42** [comment]
Can you add unit tests for these?

**Comment by andyzhangx** [comment]
@msau42  I just found there are no local volume tests for windows, I will enable those tests on windows also.

File: pkg/volume/util/util.go [hunk 2]
```diff
@@ -721,3 +721,21 @@ func CheckVolumeModeFilesystem(volumeSpec *volume.Spec) (bool, error) {
 	}
 	return true, nil
 }
+
+// MakeAbsolutePath convert path to absolute path according to GOOS
+func MakeAbsolutePath(goos, path string) string {
+	if goos != "windows" {
+		return "/" + path
```
Comment (msau42) [inline pkg/volume/util/util.go]: Can you also call filepath.Clean() on the path, so things like "//path" are simplified?

File: pkg/volume/local/local.go (hunk 1 above)
Comment (andyzhangx) [reply pkg/volume/local/local.go]: fixed

**Review by andyzhangx — COMMENTED** [review]
@msau42 I have splitted the unit tests of local volume, most of them should also run on Windows too. PTAL, thanks.

File: pkg/volume/util/util.go (hunk 2 above)
Comment (andyzhangx) [reply pkg/volume/util/util.go]: fixed

File: pkg/volume/util/util.go [hunk 3]
```diff
@@ -721,3 +722,21 @@ func CheckVolumeModeFilesystem(volumeSpec *volume.Spec) (bool, error) {
 	}
 	return true, nil
 }
+
+// MakeAbsolutePath convert path to absolute path according to GOOS
+func MakeAbsolutePath(goos, path string) string {
+	if goos != "windows" {
+		return "/" + filepath.Clean(path)
```
Comment (msau42) [inline pkg/volume/util/util.go]: Clean should be called after prepending "/".  Ie, if the user already specified an absolute path.  A test case should be added too

Comment (andyzhangx) [reply pkg/volume/util/util.go]: @msau42 thanks, fixed, and added a new unit test

**Comment by msau42** [comment]
LGTM please squash

/assign @thockin 
for approval

**Pull request merged**